src/protocols/secure_channel/PairingSession.cpp

```diff
@@ -237,7 +238,16 @@ void PairingSession::Clear()
         // middle of our handshake. In that case, there is no point in trying to
         // do MRP resends of the last message we sent. So, abort the exchange
         // instead of just closing it.
+
+        // ExchangeContext::Abort will auto-release the underlying values. Since
+        // we are using reference counted handles, we Retain to increase the
+        // reference count before Abort, so that Abort() just closes without
+        // invalidating handle reference counts
```

Comment:
The reference counted handle is supposed to be a separate ref from the ref that Abort() here is decrementing.

Reply:
Do you happen to know the code paths which use non-reference-counted handles vs the ones that the reference counted handles see?

When this piece of code triggers, I do see the reference count being `1` when this code enters, so both abort and the handle deletion tries to call Release, calling it 2x for a count of 1, which is no good.

Reply:
That's a really broad question.  When an exchange is created, it starts with a refcount of 1, and the responsibility for that refcount is transferred between the network stack and the exchange delegate in a somewhat complicated dance....